apps/aecore/src/aec_consensus_hc.erl

```diff
@@ -267,11 +267,20 @@ state_pre_transform_node(Type, Height, PrevNode, Trees) ->
     {ok, PrevHash} = aec_headers:hash_header(PrevHeader),
     {TxEnv0, _} = aetx_env:tx_env_and_trees_from_hash(aetx_transaction, PrevHash),
     TxEnv = aetx_env:set_height(TxEnv0, Height),
-    {ok, #{epoch := Epoch} = EpochInfo} = aec_chain_hc:epoch_info({TxEnv, Trees}),
+    {ok, #{epoch := Epoch, first := EpochFirst, last := EpochLast} = EpochInfo} =
+        aec_chain_hc:epoch_info({TxEnv, Trees}),
     {ok, Leader} = leader_for_height(Height, {TxEnv, Trees}),
     case Type of
         key ->
-            case Height == maps:get(last, EpochInfo) of
+            case Height =:= EpochFirst of
```

Comment:
```suggestion
            case Height of
```

Reply:
and the corresponding changes, would read better?

Reply:
Turned into ifs instead, maybe clearer.